packages/dashboard-core-plugins/src/linker/Linker.tsx

```diff
@@ -656,17 +663,31 @@ export class Linker extends Component<LinkerProps, LinkerState> {
     const { linkInProgress } = this.state;
     const { isolatedLinkerPanelId } = this.props;
 
-    // Link not started yet - no need to update type
-    if (linkInProgress?.start == null) {
-      return true;
+    if (tableColumn == null) {
+      if (linkInProgress?.start != null) {
+        // Link started, end point is not a valid target
+        this.updateLinkInProgressType(linkInProgress);
+      }
+      return false;
     }
 
-    if (tableColumn == null) {
-      // Link started, end point is not a valid target
-      this.updateLinkInProgressType(linkInProgress);
+    // TODO: Use preview/original type property when core/#3358 is completed
+    if (
+      tableColumn.description !== undefined &&
+      tableColumn.description?.startsWith('Preview of type: ')
+    ) {
```

Comment:
I'd extract this logic into a utility function (put in `LinkerUtils`), e.g.:
```
function isLinkableColumn(column: LinkColumn): boolean {
  // TODO: core/#3358 Use preview/original type property instead of checking description
  return !column.description?.startsWith('Preview of type: ');
}
```